### DISCUSSION TITLE
Can't access from LAN

### DISCUSSION THREAD

**Discussion by sksar** [pocketbase/pocketbase#4219] [Q&A]
@ganigeorgiev 

I am trying to host a pocketbase instance on LAN.
This is what I am doing 
`./pocketbase serve --http="0.0.0.0:8090"`

This works and starts the server up successfully. (I am on Mac M1)
It works when I open `http://0.0.0.0:8090/` or `http://localhost:8090/` or `http://192.168.1.157:8090/` from my the same computer.

However, when I am trying to open it up from outside (from LAN), it does not load. It just keeps spinning and says the server did not send any response. (I am trying this after turning off the Firewall on my Mac).

I also tried doing `./pocketbase serve --http="192.168.1.157:8090"` (That being the IP of my Mac)

But this one too gives the same result. It opens from the same computer, but no other computer on the same LAN.

I can access other things served from my Mac, just fine. But not pocketbase. [Pocketbase Version 0.20.7 at the time of writing]

| On My MAC | From My Phone |
| ------------- | ------------- |
| This is from the same computer <br> I have a simple `index.html` file in the `pb_public` folder. <br> <br> ![Screenshot 2024-01-23 at 16 18 41](https://github.com/pocketbase/pocketbase/assets/11752260/d8477d90-233f-45ea-85ae-42ce7ad6c581) <hr>This is the terminal  <br><br> ![Screenshot 2024-01-23 at 16 24 18](https://github.com/pocketbase/pocketbase/assets/11752260/1fd906ca-2ada-4f0b-9dc6-0df3b7b596cc) | ![photo_2024-01-23 16 18 57](https://github.com/pocketbase/pocketbase/assets/11752260/57773c48-7d3e-41c9-b82c-679d9da382e4)  |

**Comment by ganigeorgiev** [comment]
> I can access other things served from my Mac, just fine. But not pocketbase. 

What are those other things and on what address they bind?

In any case, I don't have access to a mac to test it but I don't think this is PocketBase related.

Make sure that your mac can accept inbound localhost connections (if you have a firewall it is most likely disallowed by default).

Alternatively, for accessing localhost from external devices I usually use [ngrok](https://ngrok.com/download) (`ngrok http 8090` and use the generated https address when connecting to PocketBase).

**Reply by sksar** [reply]
Other things mean, other devices such as my phone and other laptops/phones.
I think the same that, Mac could be the issue. I am currently looking at the code here, https://github.com/pocketbase/pocketbase/blob/master/apis/serve.go

Also, as I mentioned, the firewall is turned off. I started a SvelteKit project which I am easily able to access from my Phone, so there is no network level issue here.

I want to deploy this in a LAN environment. This will not be a hosted on a public server.
So I guess for now, I will have to install ngnix or caddy locally to reverse proxy. I don't see any other option for now.

**Reply by ganigeorgiev** [reply]
> I started a SvelteKit project which I am easily able to access from my Phone

I suppose you are using the `--host` vite option? Then based on [their documentation](https://vitejs.dev/config/server-options) using `--http=0.0.0.0:8090` with PocketBase should do the same.

I don't see how using a reverse proxy will change something in this case (most proxies by default also bind to `0.0.0.0` or its ipv6 equivalent).

**Reply by ganigeorgiev** [reply]
_Side-note: Also I'm not sure if this is just how the address is visualized in the phone screenshot, but double check that you are including the port when accessing the requested address._

**Reply by sksar** [reply]
Yes, I am using the --host option. Which works. I believe the `--http=0.0.0.0:8090` does the same too. However pocketbase is unable to serve the request. From the network packets tracing I can see that the connection is going, but pocketbase is not responding anything. Pocketbase is only responding to requests if they are coming in from the same machine.

**Reply by ganigeorgiev** [reply]
> Pocketbase is only responding to requests if they are coming in from the same machine

If that was the case then you wouldn't be able to access for example the public demo - https://pocketbase.io.

There is something else going on, but I'm not sure what exactly. It could be something Mac specific or some other service interfering with the request.

**Reply by sksar** [reply]
I believe it is a mac only thing. That on mac due to some reason pocketbase is not responding to outside connections.

**Reply by sksar** [reply]
I have tried different ports too. Just to be sure, I will try the same from a friend's Mac later.

**Reply by sksar** [reply]
I have caddy on my Mac. So I did this `caddy reverse-proxy --from :80  --to http://localhost:8090` to serve it from port 80. (Caddy is serving on port 80 and proxying the packets locally to pocketbase)




| Caddy running on my Mac  | My Phone |
| ---------------- | ------------- |
| This is the command  <br><br>   ![Screenshot 2024-01-23 at 17 37 33](https://github.com/pocketbase/pocketbase/assets/11752260/d0d12116-7f99-40e8-b10d-05a13bd5f7f7) <br> <br> <br> <br>  This is what caddy starts up as <br> <br> ![Screenshot 2024-01-23 at 17 37 46](https://github.com/pocketbase/pocketbase/assets/11752260/2b027d9f-d743-41aa-a123-27f42bc13914) |  From my phone it is now accessible from `http://192.168.1.157/` but not `http://192.168.1.157:8090/` <br><br>  ![IMG_6923](https://github.com/pocketbase/pocketbase/assets/11752260/cd6f83f3-abf9-4f7d-812e-6e3ebfd24ae8) |

**Reply by sksar** [reply]
This reverse proxy method works, as expected. So it is not a Network related issue I believe. I think it has to do something with Pocketbase on Mac.

**Reply by ganigeorgiev** [reply]
Contributions to fix it in case it is a PocketBase issue are welcomed but again I don't see what could be the difference from just using `--http=0.0.0.0:8090` and until proven otherwise for me this remains a userland configuration issue (it could be also a cached error response on the device from where you are testing).

**Reply by sksar** [reply]
@ganigeorgiev 
I have been trying to do this since yesterday. I tested this with 3 different smartphones. And I could not figure it out. Just to be sure, I flushed out the cache too. Even did a DHCP purge and set my Mac's IP to a different value. But I was always getting the same issue. 

I will see to this if it is my Mac that is causing the issue, and update here.
I agree with you, I do not see any difference too on why `--http=0.0.0.0:8090` does not work. It should work as is.

**Reply by fib** [reply]
I am having the same issue running PB on an Ubuntu compute instance on Oracle Cloud. Was only able to get it to work using the same Caddy configuration.
